### ISSUE TITLE
Training SequenceTagger on GPU machine, predicting on non GPU machine

### ISSUE THREAD

**Issue by JoaoMVR** [flairNLP/flair#635]
Hi, 

I've been using Flair for NER in Portuguese texts. I successfully trained a model for this task in an Azure Virtual Machine setup with GPU card. The model trained successfully and I could load it and use it in the VM.

I exported the trained model, so I could use it in my local machine, which is a MacBook Air (from 2017 if i'm not mistaken), with no CUDA supported GPU, only the integrated graphics card.

I could load the model locally, if I print it, it gives me this info:

![image](https://user-images.githubusercontent.com/16072235/55173194-2c6de800-5173-11e9-90f9-c46864374245.png)

However, when I try to use the model to tag a new sentence, with model.predict(sentence), I get the error:

**Torch not compiled with CUDA enabled**

Is it possible to use this model in a non CUDA GPU machine? Did I do something wrong when saving or loading the model?

Thanks in advance!

**Comment by JoaoMVR** [comment]
Error traceback:

![image](https://user-images.githubusercontent.com/16072235/55248315-74f3d700-5241-11e9-8f19-f783dcddac39.png)

I noticed that it points back to allennlp, and found this related issue:
https://github.com/allenai/allennlp/issues/877

However, from Flair, how can I set the cuda device to -1?

**Comment by JoaoMVR** [comment]
I managed to solve this by explicitly setting the cuda_device to -1 in the elmo.py, ElmoEmbedder class, batch_to_embeddings() function, before the cuda_device verification is made:

![image](https://user-images.githubusercontent.com/16072235/55324739-e5d20380-547a-11e9-8189-cb79d8441fab.png)

Is there a cleaner way of doing this, from Flair maybe?

**Comment by alanakbik** [comment]
Hello @JoaoMVR thanks for reporting this and sharing the solution! I think we could probably override the `.to(device)` method of the ELMoEmbedding class to always set this parameter accordingly so that this works automatically. I'll add a 'enhancement' label to this issue and will try working this in once I'm back in office (in 2 weeks)!

**Comment by alanakbik** [comment]
Hello @JoaoMVR please excuse the delay - I only now got around to looking into this. I tried running this code with an elmo model trained on GPU: 

```python
sentence = Sentence('Hello World')

tagger: TextClassifier = TextClassifier.load('elmo.pt')

elmo: ELMoEmbeddings = tagger.document_embeddings.embeddings.embeddings[0]
elmo.ee.cuda_device = -1

tagger.predict(sentence)
```

But it throws a `AttributeError: 'Conv1d' object has no attribute 'padding_mode'`. Did you encounter it as well? How did you manage to get it running on CPU? Thanks!

**Comment by JoaoMVR** [comment]
Hey @alanakbik,

I ran into that issue as well and found it to be a bug introduced in the torch library. I fixed it by explicitly using `torch = "==1.0.1"` .

**Comment by alanakbik** [comment]
Ah interesting, thanks. We should probably hold off then until either torch or allennlp fixes this.

**Comment by JoaoMVR** [comment]
I believe these are unrelated issues, but ok by me.

**Comment by falcaopetri** [comment]
@alanakbik I guess your suggestion of 
> ```python
> elmo: ELMoEmbeddings = tagger.document_embeddings.embeddings.embeddings[0]
> elmo.ee.cuda_device = -1
> ```

is related to my answer at https://github.com/flairNLP/flair/issues/610#issuecomment-584716742. Basically, I think it is because `ElmoEmbedder` stores the `cuda_device` at training time. When it gets pickled and then unpickled, `ElmoEmbedder` still uses the "cached" value of `cuda_device`.

**Comment by alanakbik** [comment]
We just put in a PR that hopefully fixes this (#1803)
